parser.js

```diff
@@ -8,189 +8,29 @@
-         */
-        if (options.errorOnUnknownASTType) {
-            extra.errorOnUnknownASTType = true;
-        }
-
-        if (typeof options.useJSXTextNode === "boolean" && options.useJSXTextNode) {
-            extra.useJSXTextNode = true;
-        }
-
-        /**
-         * Allow the user to override the function used for logging
-         */
-        if (typeof options.loggerFn === "function") {
-            extra.log = options.loggerFn;
-        } else if (options.loggerFn === false) {
-            extra.log = Function.prototype;
-        }
-
-        /**
-         * Provide the context as to whether or not we are parsing for ESLint,
-         * specifically
-         */
-        if (additionalParsingContext.isParseForESLint) {
-            extra.parseForESLint = true;
-        }
-    }
-
-    if (!isRunningSupportedTypeScriptVersion && !warnedAboutTSVersion) {
-        const border = "=============";
-        const versionWarning = [
-            border,
-            "WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-eslint-parser.",
-            "You may find that it works just fine, or you may not.",
-            `SUPPORTED TYPESCRIPT VERSIONS: ${SUPPORTED_TYPESCRIPT_VERSIONS}`,
-            `YOUR TYPESCRIPT VERSION: ${ACTIVE_TYPESCRIPT_VERSION}`,
-            "Please only submit bug reports when using the officially supported version.",
-            border
-        ];
-        extra.log(versionWarning.join("\n\n"));
-        warnedAboutTSVersion = true;
-    }
-
-    // Even if jsx option is set in typescript compiler, filename still has to
-    // contain .tsx file extension
-    const FILENAME = (extra.ecmaFeatures.jsx) ? "eslint.tsx" : "eslint.ts";
-
-    const compilerHost = {
-        fileExists() {
-            return true;
-        },
-        getCanonicalFileName() {
-            return FILENAME;
-        },
-        getCurrentDirectory() {
-            return "";
-        },
-        getDefaultLibFileName() {
-            return "lib.d.ts";
-        },
-
-        // TODO: Support Windows CRLF
-        getNewLine() {
-            return "\n";
-        },
-        getSourceFile(filename) {
-            return ts.createSourceFile(filename, code, ts.ScriptTarget.Latest, true);
-        },
-        readFile() {
-            return null;
-        },
-        useCaseSensitiveFileNames() {
-            return true;
-        },
-        writeFile() {
-            return null;
-        }
-    };
-
-    const program = ts.createProgram([FILENAME], {
-        noResolve: true,
-        target: ts.ScriptTarget.Latest,
-        jsx: extra.ecmaFeatures.jsx ? "preserve" : undefined
-    }, compilerHost);
-
-    const ast = program.getSourceFile(FILENAME);
-
-    extra.code = code;
-    return convert(ast, extra);
-}
+const parse = require("typescript-estree").parse;
+const astNodeTypes = require("typescript-estree").AST_NODE_TYPES;
+/* eslint-disable */
```

Comment:
What is this `eslint-disable` comment needed for?

Reply:
From memory I think cause I am deep importing from a node_module

Reply:
@platinumazure This was a great thing to flag because the underlying issue was that with this change ESLint went from being just a devDep to a full dep of the project, there is no need for the disable comment at all now that I have moved it into deps